### PR TITLE
Adding replace tokens for national cloud scenarios

### PR DESCRIPTION
**Pull request by aiwangmicrosoft**
Are we the only ones that use AzureCloudInstance? See changes in app.xaml.cs.

### CODE REVIEW

File: active-directory-wpf-msgraph-v2/MainWindow.xaml.cs [hunk 1]
```diff
@@ -15,7 +15,8 @@ public partial class MainWindow : Window
         //Set the API Endpoint to Graph 'me' endpoint. 
         // To change from Microsoft public cloud to a national cloud, use another value of graphAPIEndpoint.
         // Reference with Graph endpoints here: https://docs.microsoft.com/graph/deployments#microsoft-graph-and-graph-explorer-service-root-endpoints
-        string graphAPIEndpoint = "https://graph.microsoft.com/v1.0/me";
+        static string graphAPIEndpoint = "https://graph.microsoft.com";
```
Comment (jmprieur) [inline active-directory-wpf-msgraph-v2/MainWindow.xaml.cs]: You don't need to change the line into two lines, @aiwangmicrosoft 
The portal would replace https://graph.microsoft.com by the value of `msgraphEndpointHost` known by the portal.

**Review by jmprieur — CHANGES_REQUESTED** [review]
Thanks @aiwangmicrosoft 
this is a good start

We also need to change this line https://github.com/Azure-Samples/active-directory-dotnet-desktop-msgraph-v2/blob/1d85e320ae7c33978c38c0abfb52739b413ec6a8/active-directory-wpf-msgraph-v2/App.xaml.cs#L16

so that the portal can specify the cloud instance

Something like:

```CSharp
 .WithAuthority($"{Instance}{Tenant})
```

and also add a line after [line 34](https://github.com/Azure-Samples/active-directory-dotnet-desktop-msgraph-v2/blob/1d85e320ae7c33978c38c0abfb52739b413ec6a8/active-directory-wpf-msgraph-v2/App.xaml.cs#L34):

```CSharp
 private static string Instance = "https://login.microsoftonline.com/";
```

**Comment by aiwangmicrosoft** [comment]
Updated according to comments.

File: AppCreationScripts/apps.json [hunk 2]
```diff
@@ -28,15 +28,20 @@
...
+          "replaceTokens": {
+            "appId": "4a1aa1d5-c567-49d0-ad0b-cd957a47f842",
+            "tenantId": "common",
+            "authorityEndpointHost": "https://login.microsoftonline.com/"
+          }
+        },
+        {
+          "settingFile": "/active-directory-wpf-msgraph-v2/MainWindow.xaml.cs",
+          "replaceTokens": {
+            "msgraphEndpointHost": "https://graph.microsoft.com"
```
Comment (jmprieur) [inline AppCreationScripts/apps.json]: Is the portal going to replace with or without the trailing slash @archieag  ?

**Review by jmprieur — APPROVED** [review]
LGTM. @aiwangmicrosoft 
Thank you!

File: AppCreationScripts/apps.json (hunk 2 above)
Comment (archieag) [reply AppCreationScripts/apps.json]: @jmprieur The portal will replace it with a trailing slash.

Comment (jmprieur) [reply AppCreationScripts/apps.json]: @aiwangmicrosoft : then we want to have   "msgraphEndpointHost": "https://graph.microsoft.com/"

Comment (aiwangmicrosoft) [reply AppCreationScripts/apps.json]: The latest commit added the trailing slash.

**Review by jmprieur — APPROVED** [review]
Thanks @aiwangmicrosoft 
LGTM

**Pull request merged**